oauth_dropins/indieauth.py

```diff
@@ -50,6 +53,32 @@ def discover_authorization_endpoint(me):
   return INDIEAUTH_URL
 
 
+def build_user_json(me, resp=None):
+  """user_json contains an h-card, rel-me links, and "me"
+  Args:
+    me: string, URL of the user, returned by
+    resp: requests.Response (optional), re-use response if it's already been fetched
+  Return:
+    dict, with 'me', the URL for this person; 'h-card', the representative h-card
+      for this page; 'rel-me', a list of rel-me URLs found at this page
+  """
+  user_json = {'me': me}
+
+  resp = resp or util.requests_get(me)
+  if resp.status_code // 100 != 2:
+    logging.warning(
+      'could not fetch user url "%s". got response code: %d',
+      me, resp.status_code)
+    return user_json
+  p = mf2py.parse(doc=resp.text
+                  if 'charset' in resp.headers.get('content-type', '')
```

Comment:
this idiom is necessary if the character encoding is defined in the body of the html (`<meta charset="UTF-8">`). if we're not worried about encodings, and maybe we're not for this particular use case since it's almost all URLs, then that's fine.